nautobot/ipam/utils/migrations.py

```diff
@@ -308,18 +352,17 @@ def reparent_prefixes(apps):
 
     if "test" not in sys.argv:
         print("\n>>> Processing Prefix parents, please standby...")
-    for pfx in Prefix.objects.all().order_by("-prefix_length", "tenant"):
-        try:
-            parent = get_closest_parent(apps, pfx, pfx.namespace.prefixes.all())
-            if pfx.namespace != parent.namespace:
-                raise ValidationError("Prefix and parent are in different Namespaces")
-            # TODO: useful but potentially very noisy. Do migrations have a verbosity option?
-            # if "test" not in sys.argv:
-            #     print(f">>> {pfx.network}/{pfx.prefix_length} parent: {parent.network}/{parent.prefix_length}")
-            pfx.parent = parent
-            pfx.save()
-        except Prefix.DoesNotExist:
+    for pfx in Prefix.objects.all().order_by("-prefix_length", "tenant").select_related("namespace").iterator():
+        parent = get_closest_parent(pfx, pfx.namespace.prefixes.all())
+        if parent is None:
             continue
```

Comment:
nitpick: this is a bit confusing. Why not rewrite to

```py
if parent is not None:
    pfx.parent = parent
    pfx.save()
```

I don't think we would ever encounter the commented out condition `if pfx.namespace != parent.namespace` anyway, right?

Reply:
We shouldn't encounter that case, yeah, which is why I commented it out. Agree we could reverse the logic here and remove the commented part.